sdk/storage/src/common/storage_checksum.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+namespace Azure { namespace Storage { namespace Common {
+
+enum ChecksumType
+{
+    None,
+    MD5,
+    SHA256,
+    CRC64
+};
+
+class Checksum
+{
+public:
+    Checksum(ChecksumType type, std::string value) : mType(type), mValue(std::move(value)) {}
+    Checksum(ChecksumType type, uint64_t value)
+    {
+        if (type != ChecksumType::CRC64)
+        {
+            throw StorageException("Invalid checksum type");
```

Comment:
Is this a precondition check? Note that guidelines forbid throwing a C++ exception for a precondition violation: https://azure.github.io/azure-sdk/cpp_design.html